mintscript/src/script/mod.rs

```diff
@@ -121,11 +121,25 @@ impl Threshold {
     }
 }
 
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum HashType {
+    RIPEMD160,
+    SHA1,
+    SHA256,
+    HASH160,
+    HASH256,
+}
```

Comment:
A minor suggeston here:

Maybe the hash could be included in the enum arm as data.

```rust
#[derive(Copy, Clone, PartialEq, Eq, Debug)]
pub enum HashType {
    RIPEMD160([u8; 20]),
    SHA256([u8; 32]),
    // ...
}
```
(Or use a stronger types than arrays).

This way the hash size check is pushed into the decoding phase and presumably will not have to be repeated in the implementation. If the hash is a vec, the size is encoded by the vec as well as being implicitly determined by the hash type leading to duplicate information.

Is there a reason why you went for this instead?

Reply:
No specific reason. Your suggestion for enum seems better indeed